### PR TITLE
MOB-1817 Environment staging updates

### PR DESCRIPTION
**Pull request by d4r1091**
[MOB-1817](https://ecosia.atlassian.net/browse/MOB-1817)

## Context

This PR integrates the changes done in the `Core` module.

## Approach

The key updates here are the build configuration renaming, in order to match the requirement endpoints.

`Debug` -> `Staging` endpoints
`BetaDebug` -> `Staging` endpoints
`Development_TestFlight` -> `Staging` endpoints
`Development_AppCenter` -> `Staging` endpoints
`Release` -> `Production` endpoints

It also integrates the updates to make the autocomplete done via the staging endpoint bypassing the challenge.

## Tests performed

- Build and run the `EcosiaBeta` scheme => `BetaDebug` configuration
   - OUTCOME: Staging endpoint ✅ 

- Build and run the `Ecosia` scheme => `Debug` configuration
   - OUTCOME: Staging endpoint ✅ 

- Triggered AppCenter build over CircleCI `EcosiaBeta` scheme => `Development_AppCenter` configuration
   - OUTCOME: Staging endpoint ✅ 

- Triggered TestFlight build over CircleCI `EcosiaBeta` scheme => `Development_TestFlight` configuration
   - OUTCOME: Staging endpoint ✅ 

- Triggered TestFlight build over CircleCI `Ecosia` scheme => `Release` configuration
   - OUTCOME: Production endpoint ✅ 

## Other

- I could also test the autoincrement of a real release and works.
- I may need a PR to fine-tune it. Different scope though ☝️ 

🚨Last commit to be reverted. Needed to pass tests only 😉

### CODE REVIEW

File: Client.xcodeproj/project.pbxproj
Comment (ecotopian) [inline Client.xcodeproj/project.pbxproj:12354]: It's kind of counter-intuitive for me to find both words `Development` and `TestFlight` in the the name of the configuration. My brain associates the term `Development` with a `Debug` configuration, and when building for `TestFlight` usually a `Release`-like config is used. 

So for clarity, the term `Development` must be included, so that SPM in iOS-core does infer it as non-`#Production` environment? Did I catch that correctly?

Comment (d4r1091) [reply Client.xcodeproj/project.pbxproj:12354]: Thanks for the comment @ecotopian . I agree it does sound counter intuitive. I've gone through the reason of choosing this approach more thoroughly in the Core PR.
I was planning to update the README as well as I do think that indeed providing additional knowledge to it would concretely help clarifying.

File: Client/Application/AppDelegate.swift
Comment (lucaschifino) [inline Client/Application/AppDelegate.swift:39]: 🧹 💯

File: README.md
Comment (lucaschifino) [inline README.md:73]: Quick and clear 👏 

If you think it is worth it, could even link to the ADR.

**Review by lucaschifino — COMMENTED** [review]
Looking good to me 🚀 

Will just wait for the final review and approval after Core is properly updated.

File: README.md
Comment (d4r1091) [reply README.md:73]: The only reason I woulnd't want to link it is due to this repo being public and people will land into the `404` Github page 😕 . I know it's widely practiced for Jira tickets for other OSS projects but when it comes to Github repo it may be somewhat unpleasant to find into a README doc 🤔

**Review by lucaschifino — APPROVED** [review]
LGTM 🟢

**Review by ecotopian — APPROVED** [review]
Looking great now!

**Pull request merged**
